### PR TITLE
Remove `six` in `pyomo.common`

### PR DESCRIPTION
**Pull request by mrmundt**
## Summary/Motivation:
Part of the Pyomo 6.0 effort is to replace the usage of `six` with the appropriate Python 3 methods. This does just that in the `pyomo/common` directory.

## Changes proposed in this PR:
- Remove `six` in `pyomo.common`

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/common/config.py [hunk 1]
```diff
@@ -1901,10 +1895,3 @@ def _data_collector(self, level, prefix, visibility=None, docMode=False):
 
 # Backwards compatibility: ConfigDict was originally named ConfigBlock.
 ConfigBlock = ConfigDict
-
-# In Python3, the items(), etc methods of dict-like things return
-# generator-like objects.
-if six.PY3:
-    ConfigDict.keys = ConfigDict.iterkeys
-    ConfigDict.values = ConfigDict.itervalues
-    ConfigDict.items = ConfigDict.iteritems
```
Comment (jsiirola) [inline pyomo/common/config.py:1910]: If you are going to remove this, you also need to rename the `iter*` methods to be `items`, `keys,  and `values` and deprecate the `iter*` methods.

File: pyomo/common/tests/test_config.py [hunk 2]
```diff
@@ -1479,12 +1479,8 @@ def test_block_keys(self):
...
         # lists are independent
         self.assertFalse(keys is self.config['scenario'].keys())
-        if PY3:
-            self.assertIsNot(type(keys), list)
-            self.assertEqual(list(keys), ref)
-        else:
-            self.assertIs(type(keys), list)
-            self.assertEqual(keys, ref)
+        self.assertIs(type(keys), list)
+        self.assertEqual(list(keys), ref)
```
Comment (jsiirola) [inline pyomo/common/tests/test_config.py:1483]: You kept the wrong (python 2) branch of the IF here.

File: pyomo/common/tests/test_config.py [hunk 3]
```diff
@@ -1505,10 +1501,7 @@ def test_block_values(self):
 
         # list of values
         values = self.config['scenario'].values()
-        if PY3:
-            self.assertIsNot(type(values), list)
-        else:
-            self.assertIs(type(values), list)
+        self.assertIs(type(values), list)
```
Comment (jsiirola) [inline pyomo/common/tests/test_config.py:1504]: You kept the wrong (python 2) branch of the IF here.

File: pyomo/common/tests/test_config.py [hunk 4]
```diff
@@ -1526,10 +1519,7 @@ def test_block_items(self):
 
         # list of items
         items = self.config['scenario'].items()
-        if PY3:
-            self.assertIsNot(type(items), list)
-        else:
-            self.assertIs(type(items), list)
+        self.assertIs(type(items), list)
```
Comment (jsiirola) [inline pyomo/common/tests/test_config.py:1522]: You kept the wrong (python 2) branch of the IF here.

**Review by jsiirola — CHANGES_REQUESTED** [review]
This is in pretty good shape: just a few quick things to clean up.

File: pyomo/common/tests/test_config.py [hunk 5]
```diff
@@ -1479,12 +1479,8 @@ def test_block_keys(self):
         keys = self.config['scenario'].keys()
         # lists are independent
         self.assertFalse(keys is self.config['scenario'].keys())
-        if PY3:
-            self.assertIsNot(type(keys), list)
-            self.assertEqual(list(keys), ref)
-        else:
-            self.assertIs(type(keys), list)
-            self.assertEqual(keys, ref)
+        self.assertIs(type(keys), list)
```
Comment (blnicho) [inline pyomo/common/tests/test_config.py:1482]: Should this be `assertIsNot`? Seems like you switched this from the old PY3 assertion (same with lines 1504 and 1522)

File: pyomo/common/tests/test_config.py (hunk 2 above)
Comment (mrmundt) [reply pyomo/common/tests/test_config.py:1483]: Ah, maybe you can help shed some light on this. The tests fail if we run:
```
self.assertIsNot(type(keys/values/items), list)
```
They return this error (same error for all three checks on type):
```
======================================================================
FAIL: test_block_values (pyomo.common.tests.test_config.TestConfig)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/pyomo/pyomo/pyomo/common/tests/test_config.py", line 1504, in test_block_values
    self.assertIsNot(type(values), list)
AssertionError: unexpectedly identical: <class 'list'>
```

Ergo, `type(values/keys/items)` is returning as a list.

Comment (jsiirola) [reply pyomo/common/tests/test_config.py:1483]: That was testing that the `ConfigDict` class behaved like `dict`: in Python 2, `keys`, `values`, and `items` return a `list` and `iterkeys`, ` itervalues`, and `iteritems` return generator-like things.  In Python 3, they made `keys`, `values`, and `items` return the generator-like things and got rid of the `iter`* methods.  The way we mocked that up here was to define things using Python 2 style, and then if we were running Python 3, we overrode the `keys`, `values`, and `items` with their `iter`* implementations.  You removed the magic that redefined `keys`, `values`, and `items`  (old lines 1905-1910), but still need to explicitly change the API for Python 3.

This was what I was getting at in my comment above for old lines 1905-1910.

**Review by jsiirola — APPROVED** [review]

**Review by blnicho — APPROVED** [review]

**Pull request merged**